sound/arm/vc_vchi_audioserv_defs.h

```diff
@@ -78,8 +78,13 @@ typedef struct {
 // configure the write audio samples
 typedef struct {
 	uint32_t count;		// in bytes
+#if defined(CONFIG_64BIT)
```

Comment:
I think I'd prefer to remove the ifdef and use the uint32_t callbackl/callbackh in the 32-bit case (with appropriate casting in the driver).
It's ugly in either case, but I don't really like the idea of a structure that is also used by the gpu having ifdefs the gpu isn't aware of. Fortunately the gpu doesn't do anything apart from return the values, so I think I'm happy it might work...